components/ListView/ListItemImages.js

```diff
@@ -63,173 +129,233 @@ class ListItemImages extends React.Component {
+            aria-controls={`${listItem.id}-uploads`}
+            // ^ need to fix this attribute value
+            aria-hidden={!listItem.uploads.length > 0}
+            className={`${!listItem.uploads.length > 0 ? "cc-u-not-visible" : ""}`}
+          />
       );
     }
-
+    const composeStatus = () => {
+      switch (listItem.queue_status) {
+        case "WAITING":
+          return (
+            <div className="cc-c-status">
+              <div className="cc-c-status__icon">
+                <span className="pficon pficon-pending" aria-hidden="true" />
+              </div>
+              {formatMessage(messages.imageStatusWaiting)}
+            </div>
+          );
+        case "RUNNING":
+          return (
+            <div className="cc-c-status">
+              <div className="cc-c-status__icon">
+                <div className="spinner spinner-xs" />
+              </div>
+              {formatMessage(messages.imageStatusRunning)}
+            </div>
+          );
+        case "FINISHED":
+          return (
+            <div className="cc-c-status">
+              <div className="cc-c-status__icon">
+                <span className="pficon pficon-ok" aria-hidden="true" />
+              </div>
+              {formatMessage(messages.imageStatusFinished)}
+            </div>
+          );
+        case "FAILED":
+          return (
+            <div className="cc-c-status">
+              <div className="cc-c-status__icon">
+                <span className="pficon pficon-error-circle-o" aria-hidden="true" />
+              </div>
+              {formatMessage(messages.imageStatusFailed)}
+            </div>
+          );
+        case "STOPPING":
+          return (
+            <div className="cc-c-status">
+              <em className="text-muted cc-m-full-width">
+                {formatMessage(messages.imageStatusStopping)}
+              </em>
+            </div>
+          );
+        default:
+          return <></>;
+      }
+    };
     let logsSection;
     if (this.state.logsExpanded) {
-      if (this.state.fetchingLogs) {
-        logsSection = (
-          <div>
-            <div className="spinner spinner-sm pull-left" aria-hidden="true" />
-            <FormattedMessage defaultMessage="Loading log messages" />
-          </div>
-        );
-      } else logsSection = <pre>{this.state.logsContent}</pre>;
+      logsSection = <Logs logContent={this.state.logsContent} fetchingLog={this.state.fetchingLogs} />;
     }
-
     let uploadsSection;
     if (this.state.uploadsExpanded) {
-      uploadsSection = <ListItemUploads uploads={this.props.listItem.uploads} />;
+      uploadsSection = (
+        <ImagesDataList ariaLabel={formatMessage(messages.imageUploads)}>
+          {listItem.uploads.map(upload => (
+            <ListItemUploads upload={upload} key={upload.uuid} />
+          ))}
+        </ImagesDataList>
+      );
     }
+    const size = Math.round(listItem.image_size / 10000000) / 100;
```

Comment:
I think you should use increments of 1024 here. Also this rounds to the nearest 10 MB correct?

Reply:
This rounds to the nearest hundredth decimal place. 

I originally thought that it should be increments of 1024, but when I downloaded the file, I noticed that it was the same value in GB as `const size = Math.round(listItem.image_size / 10000000) / 100;`. But this was lorax, and should be confirmed when using osbuild.